lib/odbc_adapter/schema_statements.rb

```diff
@@ -10,18 +10,18 @@ def native_database_types
     # Returns an array of table names, for database tables visible on the
     # current connection.
     def tables(_name = nil)
-      stmt   = @connection.tables
-      result = stmt.fetch_all || []
-      stmt.drop
-
+      table_names = []
       db_regex = name_regex(current_database)
       schema_regex = name_regex(current_schema)
-      result.each_with_object([]) do |row, table_names|
-        next unless row[0] =~ db_regex && row[1] =~ schema_regex
-        schema_name, table_name, table_type = row[1..3]
-        next if respond_to?(:table_filtered?) && table_filtered?(schema_name, table_type)
-        table_names << format_case(table_name)
+      stmt = @connection.prepare("SHOW TABLES IN ACCOUNT")
```

Comment:
I'm just mildly curious, executing this for my personal user in Snowflake returned a very large result. Does this work only because our snowflake connection user is more limited access wise so the table count is reasonable?

Reply:
Hmmmmm, I didn't tried it yet, but according to [this commit](https://github.com/doximity/odbc_adapter/commit/88fc21e9cc48a89c9a96aae7046f6788f30681d5): 

```
use SHOW TABLES query to address "size too big" errors for long table comments
This change fixes an issue with the `tables` call erroring out if attempting to
fetch any tables that have longer than 254 character comments on them. This fix
is based on the approach used by the patterninc fork of the gem, whose PR does
a great job of explaining the issue and idea behind the solution:

> The fetch_all function is using a bound 'remarks' column that is configured to
> have a max_length of 254 characters. That max length is not configurable and the
> column cannot be unbound. So when a table with a comment longer than 254 characters
> exists in the result set, we run into the negative string size (or size too big) error.

- patterninc#9

The primary difference here is that we're using `SHOW TABLES IN ACCOUNT` query,
which should return the same set of tables as `fetch_all` was returning before,
eliminating the need for an opt-in config flag.

Snowflake docs on the SHOW TABLES query:
https://docs.snowflake.com/en/sql-reference/sql/show-tables
```

So, I suppose `@connection.prepare("SHOW TABLES IN ACCOUNT")` would return the same as the previous `stmt.fetch_all`, of course, depending on the `account` being used.